pkgs/applications/blockchains/zcash/default.nix

```diff
@@ -25,26 +26,41 @@
   zeromq,
 }:
 
+let
+  # See https://github.com/zcash/zcash/issues/6753
+  boost' = boost.overrideAttrs (old: {
+    patches = old.patches ++ [
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/zcash/zcash/v6.2.0/depends/patches/boost/6753-signals2-function-fix.patch";
+        stripLen = 0;
+        sha256 = "sha256-LSmGZkswjbT1tDEKabGq/0e4UC6iJoo/8dJLOOHGGls=";
+      })
+    ];
+  });
+in
 stdenv.mkDerivation rec {
   pname = "zcash";
-  version = "5.4.2";
+  version = "6.2.0";
 
   src = fetchFromGitHub {
     owner = "zcash";
     repo = "zcash";
     rev = "v${version}";
-    hash = "sha256-XGq/cYUo43FcpmRDO2YiNLCuEQLsTFLBFC4M1wM29l8=";
+    hash = "sha256-1jbQfVkvMLqwzj+HoGc1NPkeYWl25nsoguxV93F7DWM=";
   };
 
   patches = [
     # upstream has a custom way of specifying a cargo vendor-directory
     # we'll remove that logic, since cargoSetupHook from nixpkgs works better
     ./dont-use-custom-vendoring-logic.patch
+
+    ./add-missing-headers.patch
+    ./reorder-dependencies.patch
```

Comment:
Could you elaborate why this patch was needed?
Maybe also add a comment into the file itself for future reference.

Reply:
I have done so.